In [13]:
# Import the necessary libraries 
import pandas as pd
import sqlite3

# LOADING THE NUMBERS DATAFRAME


In [14]:
# Load the dataframe
tn_df = pd.read_csv('./zippedData/tn.movie_budgets.csv.gz')
tn_df



id  release_date                                        movie  \
0      1  Dec 18, 2009                                       Avatar   
1      2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2      3   Jun 7, 2019                                 Dark Phoenix   
3      4   May 1, 2015                      Avengers: Age of Ultron   
4      5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   
...   ..           ...                                          ...   
5777  78  Dec 31, 2018                                       Red 11   
5778  79   Apr 2, 1999                                    Following   
5779  80  Jul 13, 2005                Return to the Land of Wonders   
5780  81  Sep 29, 2015                         A Plague So Pleasant   
5781  82   Aug 5, 2005                            My Date With Drew   

     production_budget domestic_gross worldwide_gross  
0         $425,000,000   $760,507,625  $2,776,345,279  
1         $410,600,000   $241,063,875  $1,045,663,875  
2         $350,000,000    $42,762,350    $149,762,350  
3         $330,600,000   $459,005,868  $1,403,013,963  
4         $317,000,000   $620,181,382  $1,316,721,747  
...                ...            ...             ...  
5777            $7,000             $0              $0  
5778            $6,000        $48,482        $240,495  
5779            $5,000         $1,338          $1,338  
5780            $1,400             $0              $0  
5781            $1,100       $181,041        $181,041  

[5782 rows x 6 columns]

In [15]:
# Gives us a concise summary of our dataframe by giving us basic information concerning it
tn_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id                 5782 non-null   int64 
 1   release_date       5782 non-null   object
 2   movie              5782 non-null   object
 3   production_budget  5782 non-null   object
 4   domestic_gross     5782 non-null   object
 5   worldwide_gross    5782 non-null   object
dtypes: int64(1), object(5)
memory usage: 271.2+ KB


From our findings above, we can see that our dataframe has 5782 entries and 36 columns in total. We also have two data types namely object which are contained in 5 of our columns and int64 which is in 1 of our columns. 

In [16]:
# Returns the summary statistics of our dataframe
tn_df.describe()

id
count  5782.000000
mean     50.372363
std      28.821076
min       1.000000
25%      25.000000
50%      50.000000
75%      75.000000
max     100.000000

In [17]:
# Shows us the number of rows and columns in our dataframe
print('The dataframe contains:', tn_df.shape[0],'rows')
print('The dataframe contains:', tn_df.shape[1],'columns')


The dataframe contains: 5782 rows
The dataframe contains: 6 columns


# HANDLING MISSING VALUES

In [18]:
#Checking for null values
tn_df.isnull().sum()

id                   0
release_date         0
movie                0
production_budget    0
domestic_gross       0
worldwide_gross      0
dtype: int64

There are no missing values in our dataframe.

In [19]:
#Check for duplicates
tn_df.duplicated().sum()

0

We can also see that our dataframe has no duplicates.

We have no need for the release date column in our dataframe so we will drop it. 

In [20]:
# Dropping the release_date column
tn_df= tn_df.drop('release_date', axis=1)
tn_df

id                                        movie production_budget  \
0      1                                       Avatar      $425,000,000   
1      2  Pirates of the Caribbean: On Stranger Tides      $410,600,000   
2      3                                 Dark Phoenix      $350,000,000   
3      4                      Avengers: Age of Ultron      $330,600,000   
4      5            Star Wars Ep. VIII: The Last Jedi      $317,000,000   
...   ..                                          ...               ...   
5777  78                                       Red 11            $7,000   
5778  79                                    Following            $6,000   
5779  80                Return to the Land of Wonders            $5,000   
5780  81                         A Plague So Pleasant            $1,400   
5781  82                            My Date With Drew            $1,100   

     domestic_gross worldwide_gross  
0      $760,507,625  $2,776,345,279  
1      $241,063,875  $1,045,663,875  
2       $42,762,350    $149,762,350  
3      $459,005,868  $1,403,013,963  
4      $620,181,382  $1,316,721,747  
...             ...             ...  
5777             $0              $0  
5778        $48,482        $240,495  
5779         $1,338          $1,338  
5780             $0              $0  
5781       $181,041        $181,041  

[5782 rows x 5 columns]

## Cleaning the production_budget, domestic_gross and worldwide_gross columns

In [21]:
# Checking the datatypes
tn_df.dtypes

id                    int64
movie                object
production_budget    object
domestic_gross       object
worldwide_gross      object
dtype: object

Our columns represent financial information so its important for us to work with them in integer or float form so that we are able to perform mathematical operations on them. Since the three columns have object data types, we will convert them into integers.

In [22]:
# Removing the dollar signs and commas from the columns
tn_df['worldwide_gross']= tn_df['worldwide_gross'].replace('[\$,]', '', regex= True)
tn_df['production_budget']= tn_df['production_budget'].replace('[\$,]', '', regex= True)
tn_df['domestic_gross']= tn_df['domestic_gross'].replace('[\$,]', '', regex= True)

# Converting the columns into integers
tn_df['worldwide_gross']= tn_df['worldwide_gross'].astype('int64')
tn_df['production_budget']= tn_df['production_budget'].astype('int64')
tn_df['domestic_gross']= tn_df['domestic_gross'].astype('int64')

# Previewing the first five results
tn_df.head()


id                                        movie  production_budget  \
0   1                                       Avatar          425000000   
1   2  Pirates of the Caribbean: On Stranger Tides          410600000   
2   3                                 Dark Phoenix          350000000   
3   4                      Avengers: Age of Ultron          330600000   
4   5            Star Wars Ep. VIII: The Last Jedi          317000000   

   domestic_gross  worldwide_gross  
0       760507625       2776345279  
1       241063875       1045663875  
2        42762350        149762350  
3       459005868       1403013963  
4       620181382       1316721747

In [23]:
# Checking to see if there are any 0 values in our worldwide_gross column
(tn_df['worldwide_gross'] == 0).sum()

367

Seeing as we have 367 zero values in our column, we will drop these records and retain the rest.

In [24]:
# Retaining the rows where the values in the worldwide_gross is greater than zero
tn_df= tn_df[tn_df['worldwide_gross']>0]

# Checking that the zero values have been dropped
assert (tn_df['worldwide_gross']== 0).sum()==0

Now we will create a profit column that will allow us to see the profitability of each movie.



In [25]:
# Subtracting the production_budget from the worldwide_gross
tn_df['profit']= (tn_df['worldwide_gross']-tn_df['production_budget'])
# Previewing the first five
tn_df.head()
                        

<ipython-input-25-aec90ee3631b>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  tn_df['profit']= (tn_df['worldwide_gross']-tn_df['production_budget'])


id                                        movie  production_budget  \
0   1                                       Avatar          425000000   
1   2  Pirates of the Caribbean: On Stranger Tides          410600000   
2   3                                 Dark Phoenix          350000000   
3   4                      Avengers: Age of Ultron          330600000   
4   5            Star Wars Ep. VIII: The Last Jedi          317000000   

   domestic_gross  worldwide_gross      profit  
0       760507625       2776345279  2351345279  
1       241063875       1045663875   635063875  
2        42762350        149762350  -200237650  
3       459005868       1403013963  1072413963  
4       620181382       1316721747   999721747

# LOAD THE IMDB DATASET

In [26]:
import zipfile

# Extract the database file from the ZIP archive
with zipfile.ZipFile('zippedData\im.db.zip', 'r') as zip_ref:
    zip_ref.extractall('zippedData')

# Connect to the extracted SQLite database
conn = sqlite3.connect('zippedData/im.db')

query = "SELECT name FROM sqlite_master WHERE type='table';"

tables = pd.read_sql_query(query, conn)

print("Tables in the database:")
print(tables)











Tables in the database:
            name
0   movie_basics
1      directors
2      known_for
3     movie_akas
4  movie_ratings
5        persons
6     principals
7        writers


In [27]:
query = """
     SELECT *
     FROM persons
     JOIN directors
     USING (person_id)
     JOIN movie_basics
     USING (movie_id)
     JOIN movie_ratings
     USING (movie_id);
"""
imdb_df= pd.read_sql(query, conn)
imdb_df.head()

person_id    primary_name  birth_year  death_year  \
0  nm0062879  Ruel S. Bayani         NaN         NaN   
1  nm0062879  Ruel S. Bayani         NaN         NaN   
2  nm0062879  Ruel S. Bayani         NaN         NaN   
3  nm0062879  Ruel S. Bayani         NaN         NaN   
4  nm0062879  Ruel S. Bayani         NaN         NaN   

                          primary_profession   movie_id   primary_title  \
0  director,production_manager,miscellaneous  tt1592569   Paano na kaya   
1  director,production_manager,miscellaneous  tt1592569   Paano na kaya   
2  director,production_manager,miscellaneous  tt1592569   Paano na kaya   
3  director,production_manager,miscellaneous  tt1592569   Paano na kaya   
4  director,production_manager,miscellaneous  tt2057445  No Other Woman   

   original_title  start_year  runtime_minutes                  genres  \
0   Paano na kaya        2010            110.0           Drama,Romance   
1   Paano na kaya        2010            110.0           Drama,Romance   
2   Paano na kaya        2010            110.0           Drama,Romance   
3   Paano na kaya        2010            110.0           Drama,Romance   
4  No Other Woman        2011            101.0  Drama,Romance,Thriller   

   averagerating  numvotes  
0            6.4        77  
1            6.4        77  
2            6.4        77  
3            6.4        77  
4            6.4       256

In [28]:
imdb_df=imdb_df.drop(['birth_year', 'death_year', 'primary_profession', 'start_year', 'numvotes', 'primary_title', 'original_title'], axis=1)
imdb_df.head(20)

person_id       primary_name   movie_id  runtime_minutes  \
0   nm0062879     Ruel S. Bayani  tt1592569            110.0   
1   nm0062879     Ruel S. Bayani  tt1592569            110.0   
2   nm0062879     Ruel S. Bayani  tt1592569            110.0   
3   nm0062879     Ruel S. Bayani  tt1592569            110.0   
4   nm0062879     Ruel S. Bayani  tt2057445            101.0   
5   nm0062879     Ruel S. Bayani  tt2057445            101.0   
6   nm0062879     Ruel S. Bayani  tt2057445            101.0   
7   nm0062879     Ruel S. Bayani  tt2590280            100.0   
8   nm0062879     Ruel S. Bayani  tt8421806            115.0   
9   nm0062879     Ruel S. Bayani  tt8421806            115.0   
10  nm0064023      Bryan Beasley  tt4512140             53.0   
11  nm0067234       Hans Beimler  tt2098699             90.0   
12  nm0068874  Hava Kohav Beller  tt7701650            120.0   
13  nm0070482        Joel Bender  tt1016090             88.0   
14  nm0075666      Joe Berlinger  tt2028530            121.0   
15  nm0075666      Joe Berlinger  tt2112868            108.0   
16  nm0075666      Joe Berlinger  tt2481498            110.0   
17  nm0075666      Joe Berlinger  tt2481498            110.0   
18  nm0075666      Joe Berlinger  tt3137552             85.0   
19  nm0075666      Joe Berlinger  tt3326366            107.0   

                           genres  averagerating  
0                   Drama,Romance            6.4  
1                   Drama,Romance            6.4  
2                   Drama,Romance            6.4  
3                   Drama,Romance            6.4  
4          Drama,Romance,Thriller            6.4  
5          Drama,Romance,Thriller            6.4  
6          Drama,Romance,Thriller            6.4  
7                           Drama            5.8  
8                            None            7.9  
9                            None            7.9  
10                    Documentary            8.9  
11                          Drama            5.8  
12                    Documentary            5.4  
13        Horror,Romance,Thriller            5.9  
14              Crime,Documentary            8.1  
15  Biography,Documentary,History            7.8  
16          Biography,Crime,Drama            6.7  
17          Biography,Crime,Drama            6.7  
18     Biography,Documentary,News            6.5  
19    Biography,Crime,Documentary            6.8

In [29]:
imdb_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 181387 entries, 0 to 181386
Data columns (total 6 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   person_id        181387 non-null  object 
 1   primary_name     181387 non-null  object 
 2   movie_id         181387 non-null  object 
 3   runtime_minutes  163584 non-null  float64
 4   genres           180047 non-null  object 
 5   averagerating    181387 non-null  float64
dtypes: float64(2), object(4)
memory usage: 8.3+ MB


In [30]:
imdb_df.describe()

runtime_minutes  averagerating
count    163584.000000  181387.000000
mean         97.789484       6.217683
std         194.434689       1.388026
min           3.000000       1.000000
25%          84.000000       5.400000
50%          94.000000       6.300000
75%         107.000000       7.200000
max       51420.000000      10.000000

In [31]:
imdb_df.shape

(181387, 6)

In [32]:
imdb_df.isnull().sum()

person_id              0
primary_name           0
movie_id               0
runtime_minutes    17803
genres              1340
averagerating          0
dtype: int64

In [33]:
imdb_df.dropna(subset=['runtime_minutes', 'genres'], inplace= True)


In [34]:
imdb_df['runtime_minutes']= imdb_df['runtime_minutes'].astype('int64')

In [35]:
imdb_df.dtypes

person_id           object
primary_name        object
movie_id            object
runtime_minutes      int64
genres              object
averagerating      float64
dtype: object